### ISSUE TITLE
Violin / boxplots should behave like groupedbar w/r/t groups

### ISSUE THREAD

**Issue by kescobo** [JuliaPlots/StatPlots.jl#145]
Not sure how difficult this would be, but violin and boxplot groups would make far more sense if the default were analogous to `bar_position=:dodge` for grouped bars (it's the default for `groupedbar()` atm). 

```
using StatPlots

df = DataFrame(x=repmat([1,2,3],4), y=rand(12), g=vcat(fill("a", 6), fill("b", 6)))
@df df groupedbar(:x, :y, group=:g)
```

![bar](https://user-images.githubusercontent.com/3502975/39387824-dc2280f8-4a49-11e8-8352-962b1bb317e6.png)

```
@df df violin(:x, :y, group=:g)
```

![violin](https://user-images.githubusercontent.com/3502975/39387830-e4f0bc9a-4a49-11e8-8136-c164d2df204d.png)

```
@df df boxplot(:x, :y, group=:g)
```

![box](https://user-images.githubusercontent.com/3502975/39387836-ef9d87ea-4a49-11e8-8b47-d320e3e06766.png)

I looked at the code for the `groupedbar()` user recipe, and it seems fairly straightforward, but the code for `violin()` and `boxplot()` was too hard for me to grok at 6pm on a friday. My main hesitation about wading into them is that it looks like they're not user recipes the way that `groupedbar()` is, and I don't know the implications of converting them.

**Comment by piever** [comment]
If you make a PR I'd be happy to review it, this thing has been on the to do list for a while. It's not completely straightforward, what I would recommend is to start making some `groupedboxplot` and `groupedviolin` that behave as you want as user recipes (just like `groupedbar`). I believe the best way to give them data is probably as two tuples of vectors (one tuple with the x values for each group and another with the y values for each group): the matrix approach here makes less sense as different columns may have different lengths. You could try copying the `groupedbar` approach as much as possible but each series should be `:boxplot` rather than `:bar`.

Then we'll see how to make it so that `groupedboxplot` with a grouping argument calls `groupedboxplot` with the tuples (we should do a similar trick as with `groupedbar`, but shouldn't be too difficult).

**Comment by kescobo** [comment]
I agree with @mkborregaard - the main barrier is the introduction of a new keyword right? In the case of grouped bar, it's the `bar_position` keyword, though I'm not actually sure that would be necessary for violin and boxplots (I've never seen the `stacked` form).

**Comment by mkborregaard** [comment]
@kescobo I agree with you, if you just implement a `:dodge` version we won't need a keyword to turn it off. Very happy to review and merge a PR on this, it's a good idea.

**Comment by kescobo** [comment]
@mkborregaard Thanks for resurfacing this. I've been doing RCall to get these plots and it's miserable. I don't know when I'll find the time, but will see what I can do :-)

**Comment by mkborregaard** [comment]
This is actually a duplicate of #89 - Since there's more discussion here I'm going to close the older issue and keep this open but it's worth to read the talk there.

**Comment by piever** [comment]
Closed by #192
